### PR TITLE
Alternative rcache fix

### PR DESCRIPTION
**Pull request by artpol84**
The issues with #5:
1. Sub-dt UMRs are not tracked
2. The assumption that all iov's have same memh is not correct [[link](https://github.com/brminich/ucx/pull/5/files#diff-ad9bdaed881e36d0aa5f7b36a73599fdR392)]

### CODE REVIEW

**Comment by artpol84** [comment]
@brminich  Work In Progress - do not merge

File: src/ucp/dt/dt_struct.h [hunk 1]
```diff
@@ -51,6 +53,7 @@ typedef struct ucp_dt_struct {
     size_t len, step_len, depth;
     size_t desc_count;
     size_t rep_count;
+    ucp_dt_reg_t contig_mem;
```
Comment (brminich) [inline src/ucp/dt/dt_struct.h]: would need to save it in the `state`, struct does not represent a single registration - the same struct can be registered for different memory addresses

Comment (artpol84) [reply src/ucp/dt/dt_struct.h]: Yes, I figured that out. Haven’t yet implemented

Comment (artpol84) [reply src/ucp/dt/dt_struct.h]: See 840ac54 on how I was planning to approach this

**Comment by artpol84** [comment]
These fixes were integrated into #6

**Pull request closed without merging**